libraries/WiFi/src/WiFiUdp.cpp

```diff
@@ -0,0 +1,105 @@
+#include "WiFiUdp.h"
+#include <Arduino.h>
+
+#define udp_assert(cy_ret)   if (cy_ret != CY_RSLT_SUCCESS) { \
+            _status = SOCKET_STATUS_ERROR; \
+            return; \
+}
+
+#define udp_assert_raise(cy_ret)   if (cy_ret != CY_RSLT_SUCCESS) { \
+            return cy_ret; \
+}
+
+WiFiUDP::WiFiUDP() :
+    client_handle(nullptr),
+    _status(SOCKET_STATUS_UNINITED),
+    _last_error(CY_RSLT_SUCCESS),
+    remote_ip(0, 0, 0, 0),
+    _port(0) {
+}
+
+uint8_t WiFiUDP::begin(uint16_t port) {
+    _port = port;
+
+    // Initialize the socket for UDP
+    socket.begin(true); // true = UDP
+    if (socket.status() != SOCKET_STATUS_CREATED) {
+        return 0; // Return 0 if socket creation fails
+    }
+    // Bind the socket to the specified port
+    socket.bind(port);
+    if (socket.status() != SOCKET_STATUS_BOUND) {
+        return 0; // Return 0 if binding fail
+
+    }
+    return socket.status(); // Return the socket status
+}
+
+
+void WiFiUDP::stop() {
+
+}
+
+int WiFiUDP::beginPacket(IPAddress ip, uint16_t port) {
+    remote_ip = ip;
+    _port = port;
+    return 1; // Return 1 if successful
```

Comment:
Shouldn´t this do socket.begin()?

Reply:
No begin initialises the socket and binds to the local port, but the begin Packet function starts a connection to write UDP data to the remote connection. So calling begin inside this will reinitialise the socket every time, which is not required.

Beginpacket ( ), write( ) , endPacket() in this order.

Reply:
Okay then. I thought that begin() was only required to open a port to listen (bind).